Extensions/AzureExp/Src/Tasks/ManageExperimentV1/main.ts

```diff
@@ -9,17 +9,33 @@ async function run() {
         let serviceConnectionId = tl.getInput('ServiceConnectionId', true);
         let featureId = tl.getInput('FeatureId', true);
         let progressionId = tl.getInput('ProgressionId', true);
-        let experimentName = tl.getInput('ExperimentName', true);
         let action = tl.getInput('Action', true);
-
+        let experimentName = ExperimentAction[action] !== ExperimentAction.StopAllExperiments ? tl.getInput('ExperimentName', true) : '';
+        
         if (!ExperimentAction[action]) {
             throw new Error(tl.loc('InvalidAction', action));
         }
 
-        let experimentmanager = new ExperimentManager(featureId, progressionId, serviceConnectionId, tl.getVariable("AZURE_HTTP_USER_AGENT"));
-        let experiment = await experimentmanager.getExperiment(experimentName);
+        let experimentmanager = new ExperimentManager(featureId, progressionId, serviceConnectionId, tl.getVariable('AZURE_HTTP_USER_AGENT'));
+        let experiments = await experimentmanager.getExperiments(experimentName);
+
+        if (ExperimentAction[action] === ExperimentAction.StopAllExperiments) {
+            for (let experiment of experiments) {
+                if (!ExperimentManager.IsExperimentStopped(experiment)) {
+                    await experimentmanager.executeAction(ExperimentAction.Stop, experiment);
+                }
+                else {
+                    console.log(tl.loc('ExperimentAlreadyStopped', experiment.Name));
+                }
+            }
+        }
+        else {
+            if (experiments.length > 1) {
+                throw new Error(tl.loc('MoreThanOneExperimentFound', experimentName));
+            } 
 
-        await experimentmanager.executeAction(experiment.Id, ExperimentAction[action]);
+            await experimentmanager.executeAction(ExperimentAction[action], experiments[0]);
```

Comment:
Can you please handle the case where no experiment found? Currently it will throw nullreference exception.